crates/router/src/scheduler/consumer.rs

```diff
@@ -82,14 +84,17 @@ pub async fn start_consumer(
                     workflow_selector,
                 ));
             }
-            Ok(()) | Err(oneshot::error::TryRecvError::Closed) => {
+            Ok(()) | Err(mpsc::error::TryRecvError::Disconnected) => {
                 logger::debug!("Awaiting shutdown!");
+                ////can panic!
+                rx.close();
                 shutdown_interval.tick().await;
                 let active_tasks = consumer_operation_counter.load(atomic::Ordering::Acquire);
 
                 match active_tasks {
                     0 => {
                         logger::info!("Terminating consumer");
+                        rx.close();
```

Comment:
This won't be required, as the channel has been closed already

Reply:
On it